Formula/container-structure-test.rb

```diff
@@ -0,0 +1,25 @@
+class ContainerStructureTest < Formula
+  desc "Validate the structure of your container images"
+  homepage "https://github.com/GoogleContainerTools/container-structure-test"
+  url "https://github.com/GoogleContainerTools/container-structure-test.git",
+      :tag      => "v1.8.0",
+      :revision => "19abf36d1451cb27f8e0f5ec8260815c73184bd4"
+  head "https://github.com/GoogleContainerTools/container-structure-test.git"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    dir = buildpath/"src/github.com/GoogleContainerTools/container-structure-test"
+    dir.install buildpath.children - [buildpath/".brew_home"]
+    cd dir do
+      system "make"
+      bin.install "out/container-structure-test"
+      prefix.install_metafiles
+    end
+  end
+
+  test do
+    shell_output("#{bin}/container-structure-test version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.